app/services/fast_track/hypertension_pdf_generator.rb

```diff
@@ -2,31 +2,43 @@
 
 module FastTrack
   class HypertensionPdfGenerator
-    attr_accessor :patient, :bp_data, :medications
+    attr_accessor :patient, :blood_pressure_data, :medications, :pdf
 
-    def initialize(patient, bp_data, medications)
+    def initialize(patient, blood_pressure_data, medications)
+      @pdf = Prawn::Document.new
       @patient = patient
-      @bp_data = bp_data
+      @blood_pressure_data = blood_pressure_data
       @medications = medications
       @date = Time.zone.today
     end
 
+    # progressively builds a pdf and is sensitive to sequence
     def generate
-      pdf = Prawn::Document.new
-      pdf = add_intro(pdf)
-      pdf = add_blood_pressure(pdf)
-      pdf = add_medications(pdf) if medications.length > 1
-      add_about(pdf)
+      add_intro
+      add_blood_pressure_intro
+      add_blood_pressure_list
+      add_blood_pressure_outro
+      add_medications_intro
+      add_medications_list
+      add_about
+
+      pdf
     end
 
-    def stringify_patient
-      names = patient.with_indifferent_access
-      full_name = [names[:first], names[:middle], names[:last]].reject(&:blank?).join ' '
-      [full_name, names[:suffix]].reject(&:blank?).join ', '
+    private
+
+    def blood_pressure_data?
+      @blood_pressure_data.length.positive?
     end
 
-    def add_intro(pdf)
-      patient_name = stringify_patient
+    def medications?
+      @medications.length > 1
```

Comment:
This is interesting and raises a _slightly out of scope question_ for me about the logic here. @tadhg-ohiggins do we want this to be `>` or `=>`? Right now `medications?` only returns true if there's more than one medication but not if there's only one.

Reply:
Cool - @allthesignals do you mind updating that in a follow up commit with it's own message?

Reply:
@NatashaJIbrahim yes will do!